src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs

```diff
@@ -84,29 +84,47 @@ public class VsCredentialProviderImporter
         /// <returns>An enumeration of plugin providers</returns>
         public ICredentialProvider GetProvider()
         {
-            this._initializer();
             ICredentialProvider result = null;
 
-            if (ImportedProvider != null)
+            try
             {
-                result = new VsCredentialProviderAdapter(ImportedProvider);
-            }
+                this._initializer();
```

Comment:
I'm ok with won't fix, but I would appreciate another cleanup commit that goes just into dev. If we don't cleanup as we go through code it just never gets prioritized.

Not going to push hard, will leave it up to you